converter.js

```diff
@@ -1,104 +1,83 @@
 
 const Units = Object.freeze({
-    BU: Symbol("bu"), //Base Unit for this system
-    CM: Symbol("cm"),
-    INCH: Symbol("in"),
-    KM: Symbol("km"),
-    MT: Symbol("mt")
+    BU: "bu", //Base Unit for this system
+    CM: "cm",
+    IN: "in",
+    KM: "km",
+    MT: "mt",
+    YD: "yd"
 });
 
-class tuple {
-    #key = undefined;
-    #value = undefined;
+const units = [Units.CM, Units.IN, Units.KM, Units.MT, Units.YD];
 
-    constructor(key, value) {
-        this.#key = key;
-        this.#value = value;
-    }
-
-    get source() {
-        return `${this.#key.description}`;
-    }
-
-    get target() {
-        return `${this.#value.description}`;
-    }
-}
+const unitsOfLength = [
+    { "source": Units.BU, "target": Units.CM, "ratio": 1 },
```

Comment:
Is the concept of `BU` relevant anymore? In what situation do we expect it to come up? For example, making the following changes, does not affect the service:
```diff,js
-    { "source": Units.BU, "target": Units.CM, "ratio": 1 },
+    { "source": Units.CM, "target": Units.CM, "ratio": 1 },  // source == target, ratio==1 implies BU
```
...
```diff
-    let us = configureUnitSystem(Units.BU, 1, Units.CM);
+    let us = configureUnitSystem(Units.CM, 1, Units.CM);
```

Reply:
I agree. Currently offered features of the system doesn't require one. However, if the system is extended to support mathematical operations across different units like `2{in} * 3{mm}`, this indirection might come handy. It's not going to give us anything different or extra, but in code we can use the `bu` as a constant, which invariably refers to the `base` unit. I reiterate, it's not going to provide anything extra, but just a comfortable convention.

And in that case the following data unit might be changed to 2 parts. From the following code snippet:
```js
function configureUnitSystem(baseunit, ratio, basis) {
    const nodeGraph = { BASEUNIT: baseunit, RATIO: ratio, BASIS: basis, CHILDREN: new Map() };
    ...
```
to
```js
function configureUnitSystem(baseunit, ratio, basis) {
    const nodeGraph = { RATIO: ratio, BASIS: basis, CHILDREN: new Map() };
    BASEUNIT = baseunit;
    ...
```

Further references of `BASEUNIT` is done from a property of the closure instead of a property of `nodeGraph`.

Again this was what I thought, it will be great if you can provide anything otherwise.

Reply:
It is fine, it is just redundant information. I would be more prone to adding complexity when needed, rather than planning ahead. Although admittedly, this isn't that complicated, so it can be left as is, with the addition of the code you proposed. I think it is okay to have a "thing" we call base unit, in order to maintain consistency in future.